### PR TITLE
feat(radarr): media management

### PR DESCRIPTION
**Pull request by Fuochi**
- add mediamanagement
- add naming
- add rootfolder missing methods

### CODE REVIEW

File: radarr/mediamanagement.go [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+	return &output, nil
+}
+
+// UpdateMediaManagement updates the mediaManagement.
+func (r *Radarr) UpdateMediaManagement(mMgt *MediaManagement) (*MediaManagement, error) {
+	return r.UpdateMediaManagementContext(context.Background(), mMgt)
+}
+
+// UpdateMediaManagementContext updates the mediaManagement.
+func (r *Radarr) UpdateMediaManagementContext(ctx context.Context, mMgt *MediaManagement) (*MediaManagement, error) {
```
Comment (davidnewhall) [inline radarr/mediamanagement.go:62]: maybe `input` instead of `mMgt`?

File: radarr/rootfolder.go [hunk 2]
```diff
@@ -1,29 +1,33 @@
...
+	FreeSpace       int64         `json:"freeSpace,omitempty"`
 	Path            string        `json:"path"`
-	Accessible      bool          `json:"accessible"`
-	FreeSpace       int64         `json:"freeSpace"`
-	UnmappedFolders []*starr.Path `json:"unmappedFolders"`
+	UnmappedFolders []*starr.Path `json:"unmappedFolders,omitempty"`
 }
 
-// GetRootFolders returns all configured root folders.
+// GetRootFolders returns all configured rootFolders.
```
Comment (davidnewhall) [inline radarr/rootfolder.go:25]: "root folders."

File: radarr/mediamanagement_test.go [hunk 3]
```diff
@@ -0,0 +1,157 @@
...
+  "id": 1
+}`
+
+func TestGetMediaManagement(t *testing.T) {
+	t.Parallel()
+
+	tests := []*starr.TestMockData{
+		{
+			Name:           "200",
+			ExpectedPath:   path.Join("/", starr.API, radarr.APIver, "config/mediaManagement"),
```
Comment (davidnewhall) [inline radarr/mediamanagement_test.go:41]:
```suggestion
			ExpectedPath:   path.Join("/", starr.API, radarr.APIver, "config", "mediaManagement"),
```

and the rest..?

File: radarr/rootfolder.go [hunk 4]
```diff
@@ -1,29 +1,33 @@
...
 func (r *Radarr) GetRootFolders() ([]*RootFolder, error) {
 	return r.GetRootFoldersContext(context.Background())
 }
 
-// GetRootFoldersContext returns all configured root folders.
+// GetRootFoldersContext returns all configured rootFolders.
```
Comment (davidnewhall) [inline radarr/rootfolder.go:30]:
```suggestion
// GetRootFoldersContext returns all configured root folders.
```

**Review by davidnewhall — CHANGES_REQUESTED** [review]

**Comment by Fuochi** [comment]
while testing this I noticed I've been to aggressive with the `omitempty` tag, especially for bool fields. I have to review this, I'll probably create a PR to fix across different resources

File: radarr/rootfolder.go [hunk 5]
```diff
@@ -34,3 +38,57 @@ func (r *Radarr) GetRootFoldersContext(ctx context.Context) ([]*RootFolder, erro
 
 	return output, nil
 }
+
+// GetRootFolder returns a single root folder.
+func (r *Radarr) GetRootFolder(folderID int) (*RootFolder, error) {
```
Comment (davidnewhall) [inline radarr/rootfolder.go:43]: folder ID is an int64

File: radarr/mediamanagement.go (hunk 1 above)
Comment (Fuochi) [reply radarr/mediamanagement.go:62]: I'm not a huge fan of non-meaningful input names, I used the same naming of sonarr. If you prefer, I can expand this as either `mediaManagement` or just `management`

File: radarr/rootfolder_test.go [hunk 6]
```diff
@@ -0,0 +1,226 @@
...
+		},
+	}
+
+	for _, test := range tests {
+		test := test
+		t.Run(test.Name, func(t *testing.T) {
+			t.Parallel()
+			mockServer := test.GetMockServer(t)
+			client := radarr.New(starr.New("mockAPIkey", mockServer.URL, 0))
+			output, err := client.GetRootFolder(test.WithRequest.(int))
```
Comment (davidnewhall) [inline radarr/rootfolder_test.go:129]: also int64

File: radarr/mediamanagement.go (hunk 1 above)
Comment (davidnewhall) [reply radarr/mediamanagement.go:62]: it's fine as-is

**Comment by davidnewhall** [comment]
just a couple int->int64 fixes...

**Comment by Fuochi** [comment]
> just a couple int->int64 fixes...

yes sorry, I was still looking at the `omitempty` tag, I'll try to fix this either today or tomorrow

**Review by davidnewhall — APPROVED** [review]

**Comment by Fuochi** [comment]
Just rebased, and fixed omitepty tags again, I thought I made a mess with the tags, but in the end only few fields were actually wrongly tagged (I updated the sonarr ones as well)

File: radarr/rootfolder.go [hunk 7]
```diff
@@ -34,3 +38,57 @@ func (r *Radarr) GetRootFoldersContext(ctx context.Context) ([]*RootFolder, erro
...
+
+// DeleteRootFolderContext removes a single root folder.
+func (r *Radarr) DeleteRootFolderContext(ctx context.Context, folderID int64) error {
+	req := starr.Request{URI: path.Join(bpRootFolder, fmt.Sprint(folderID))}
+	if err := r.DeleteAny(ctx, req); err != nil {
+		return fmt.Errorf("api.Delete(%s): %w", &req, err)
+	}
+
+	return nil
+}
```
Comment (davidnewhall) [inline radarr/rootfolder.go:94]: Are these file's changes intentional? As in, did you mean to add rootFolder changes in this pull request?

**Review by davidnewhall — APPROVED** [review]
This seems just fine. Thank you very much!

**Pull request merged**